pydash/pydash_app/user/verification.py

```diff
@@ -0,0 +1,55 @@
+import uuid
+from . import repository
+import pydash_logger
+from multi_indexed_collection import DuplicateIndexError
+
+
+logger = pydash_logger.Logger(__name__)
+
+
+def verify(verification_code):
+    """
+    Attempts to verify a user with the provided verification code.
+    This is intended as a one-time action per user after registration.
+    :param verification_code: The verification code that should match the User-entity's verification code.
+        Can be a string or UUID object.
+    :return: Returns True if both verification codes are equal, returns False otherwise.
+        Raises an InvalidVerificationCodeError when the provided verification code is invalid.
+        Raises an VerificationCodeExpiredError when the provided verification code has expired.
+    """
+    # Ensure verification code can be a string, an integer or a UUID object.
+    if not isinstance(verification_code, uuid.UUID):
+        try:
+            verification_code = uuid.UUID(verification_code)
+        except ValueError:
+            # Invalid input format
+            logger.warning(f"Verification code {verification_code} is invalid - Invalid input format.")
+            raise InvalidVerificationCodeError(f"Verification code {verification_code} is invalid.")
```

Comment:
Nice use of Exceptions :+1: :smile: Are you handling them gracefully somewhere?

Reply:
Yep, in the corresponding controller file.

Reply:
Must've missed that, nice job!